Update to Roaster version which is compatible with Java 8

Trying to build `base` with Roaster v 2.23.0.Final causes the following error:

```
    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ':annotateProto'.
    > org/jboss/forge/roaster/_shade/org/eclipse/jface/text/Document has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0

    * Try:
    Run with --info or --debug option to get more log output. Run with --scan to get full insights.

    * Exception is:
    org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':annotateProto'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:188)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$$Lambda$1048/81383731.accept(Unknown Source)

Caused by: java.lang.UnsupportedClassVersionError: org/jboss/forge/roaster/_shade/org/eclipse/jface/text/Document has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
        at org.jboss.forge.roaster.spi.JavaParserImpl.parseUnit(JavaParserImpl.java:58)
        at org.jboss.forge.roaster.Roaster.parse(Roaster.java:260)
        at org.jboss.forge.roaster.Roaster.parse(Roaster.java:285)
...
```
It looks JRoaster versions after `2.22.2.Final` use shaded version of JFace build with Java 11, which causes projects using Java 8 fail, when it comes to invoking shaded classes.

When we migrate to Java 11 or higher the issue would resolve by itself. Until then we should check for compatibility of Roaster “guts” with Java 8 before we bump its version, OR stick to `2.22.2.Final`.

--- a/buildSrc/src/main/kotlin/io/spine/internal/dependency/Roaster.kt
+++ b/buildSrc/src/main/kotlin/io/spine/internal/dependency/Roaster.kt
@@ -27,17 +27,10 @@
 package io.spine.internal.dependency
 
 // https://github.com/forge/roaster
-@Suppress("unused")
+@Suppress("unused", "ConstPropertyName")
 object Roaster {
 
-    /**
-     * Do not advance this version further because it would break compatibility with Java 8
-     * projects. Starting from the following version Roaster has a shaded version of Eclipse JFace
-     * built with Java 11.
-     *
-     * Please see [this issue][https://github.com/SpineEventEngine/config/issues/220] for details.
-     */
-    private const val version = "2.24.0.Final"
+    private const val version = "2.28.0.Final"
 
     const val api = "org.jboss.forge.roaster:roaster-api:${version}"
     const val jdt = "org.jboss.forge.roaster:roaster-jdt:${version}"

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.2-bin.zip
 networkTimeout=10000
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

